pandas/io/sql.py

```diff
@@ -659,6 +668,7 @@ def read_sql(
                 parse_dates=parse_dates,
                 chunksize=chunksize,
                 use_nullable_dtypes=use_nullable_dtypes,
+                dtype=dtype,
```

Comment:
How does this interact with `use_nullable_dtypes` when `dtype` is not a nullable type?

Reply:
We should add tests for this :)

Right now we are casting to nullable and then apply astype (that's how it was implemented before as well). But I don't think that we have any tests covering this

Reply:
Yeah would be good to add tests for this case. IIRC `read_csv` or a similar reader had a similar thing. This should hopefully operate similarly

Reply:
added